modules/jdbc/src/main/java/org/testcontainers/jdbc/ContainerDatabaseDriver.java

```diff
@@ -91,9 +103,25 @@ public synchronized Connection connect(String url, final Properties info) throws
                  */
                 ServiceLoader<JdbcDatabaseContainerProvider> databaseContainers = ServiceLoader.load(JdbcDatabaseContainerProvider.class);
                 for (JdbcDatabaseContainerProvider candidateContainerType : databaseContainers) {
-                    if (candidateContainerType.supports(connectionUrl.getDatabaseType())) {
-                        container = candidateContainerType.newInstance(connectionUrl);
-                        container.withTmpFs(connectionUrl.getTmpfsOptions());
+                	
+                	String type = connectionUrl.getDatabaseType();
+                	boolean wasAlias = false;
+                	Map<String, String> aliasMap = new HashMap<String, String>();
+                	if (jdbcAliasMap.containsKey(type)) { //this is an alias
+                		aliasMap = jdbcAliasMap.get(type);
+                		type = aliasMap.get("type"); //overwrite type from url with type from alias definition
+                		wasAlias = true;
+                	}
+                	
+                    if (candidateContainerType.supports(type)) {
+                    	
+                    	container = candidateContainerType.newInstance(connectionUrl);
+                    	
+                    	if (wasAlias) { //overwrite image and tag
+                    		container.setDockerImageName(aliasMap.get("image")+":"+getAliasTag(aliasMap, container));                        	
```

Comment:
This does not feel like the right place to do it. Consider handling aliases in `JdbcDatabaseContainerProvider#newInstance`. We may also need to introduce a new API there to handle new instance creation with custom image

Reply:
@bsideup 
newInstance() is overridden in implementing modules, I didn't want to touch existing modules.
For me it's the smartest way to overwrite the imageName right after instance creation.

Reply:
this is why I was talking about introducing a new API. `setDockerImageName` here is a bit hacky and we should expose first-class support for creating JDBC containers from a custom image

Reply:
@bsideup the Container is created according to the property DockerImageName. The property as a public setter isn't that an API ;-)